### PR TITLE
front: migrate chartHelpers files to ts

### LINKED ISSUE #4956: front: migrate SimulationResults/ChartHelpers files to typescript
### Description and goal

Migrate the following files to typescript :
- [x] `drawCurve.jsx`
- [x] `drawElectricalProfile.jsx`
- [x] `drawPowerRestriction.jsx`
- [x] `svgDefs.jsx`

### PR DESCRIPTION
**Pull request by SarahBellaha**
Closes #4956

### CODE REVIEW

File: front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts [hunk 1]
```diff
@@ -1,35 +1,44 @@
...
-  name,
-  rotate,
-  isStripe,
-  isIncompatible,
-  // TODO : when migrated id should be reverted with null by default
-  id
+  chart: Chart,
+  classes: string,
+  dataSimulation: ElectricalConditionSegment,
+  groupID: string,
```
Comment (alexandredamiron) [inline front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts:15]: This is the right time to not use generic naming when this is not necessary
electricalConditionSegment: ElectricalConditionSegment

File: front/src/applications/operationalStudies/consts.ts [hunk 2]
```diff
@@ -217,7 +218,10 @@ interface Profile {
   isStriped: boolean;
 }
 
-interface ElectricalConditionSegment {
+export const DRAWING_KEYS: (keyof HeightPosition)[] = ['position', 'height'];
+export type DrawingKeys = typeof DRAWING_KEYS;
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/consts.ts:224]: Please proving some explantation about this key

File: front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+  electricalConditionSegment: ElectricalConditionSegment,
+  groupID: string,
+  keyValues: DrawingKeys,
+  rotate: boolean,
+  isStripe: boolean,
+  isIncompatible: boolean,
+  id: string
+) => {
+  const ELECTRIFIED = 'Electrified';
+  const NEUTRAL = 'Neutral';
```
Comment (SharglutDev) [inline front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts]: You could move these 2 consts before the function or in consts.ts

File: front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    start: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_start`],
+    middle: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_middle`],
+    end: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_end`],
+  };
+  // y
+  const yValues = {
+    start: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_start`],
+    middle: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_middle`],
+    end: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_end`],
+  };
```
Comment (SharglutDev) [inline front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts]: Nice idea ! Just a suggestion but as it's also used in drawPowerRestriction, maybe you could make a function to get these values and put it in chartHelpers.ts for example. The suggestion below can probably be improved.
```suggestion
 export const getValues = (
  segment: ElectricalConditionSegment | PowerRestrictionSegment,
  keyValues: DrawingKeys,
  rotate: boolean,
  axe: 'x' | 'y'
) => {
  const rotateTrueValue = axe === 'x' ? 1 : 0;
  const rotateFalseValue = axe === 'x' ? 0 : 1;

  return {
    start: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_start`],
    middle: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_middle`],
    end: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_end`],
  };
};

  const xValues = getValues(electricalConditionSegment, keyValues, rotate, 'x');
  const yValues = getValues(electricalConditionSegment, keyValues, rotate, 'y');

```

File: front/src/modules/simulationResult/components/ChartHelpers/drawPowerRestriction.ts [hunk 5]
```diff
@@ -1,36 +1,37 @@
...
+    start: powerRestrictionSegment[`${keyValues[rotate ? 1 : 0]}_start`],
+    middle: powerRestrictionSegment[`${keyValues[rotate ? 1 : 0]}_middle`],
+    end: powerRestrictionSegment[`${keyValues[rotate ? 1 : 0]}_end`],
+  };
+  // y
+  const yValues = {
+    start: powerRestrictionSegment[`${keyValues[rotate ? 0 : 1]}_start`],
+    middle: powerRestrictionSegment[`${keyValues[rotate ? 0 : 1]}_middle`],
+    end: powerRestrictionSegment[`${keyValues[rotate ? 0 : 1]}_end`],
+  };
```
Comment (SharglutDev) [inline front/src/modules/simulationResult/components/ChartHelpers/drawPowerRestriction.ts]: Same as previous file

**Review by SharglutDev — COMMENTED** [review]
Tested in local, very nice job ! Just a few suggestions

**Review by SharglutDev — APPROVED** [review]
Lgtm :)

**Review by Math-R — APPROVED** [review]
LGTM

**Pull request merged**